pull request in optimization section

zhengqu opened the pull request.

Commit:
d

Commit:
d

philipphennig commented:
Dear Zhengqu,

thanks for your pull request. You didn't provide a commit summary; could you briefly explain why you think your paper is relevant to the Probabilistic Numerics community?

Best wishes,

Philipp Hennig

> On 26 Feb 2015, at 15:02, zhengqu notifications@github.com wrote:
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/philipphennig/probabilistic-numerics.org/pull/1
> 
> Commit Summary
> 
>   • d
>   • d
> File Changes
> 
>   • M _bibliography/Optimization.bib (19)
> Patch Links:
> 
>   • https://github.com/philipphennig/probabilistic-numerics.org/pull/1.patch
>   • https://github.com/philipphennig/probabilistic-numerics.org/pull/1.diff
> —
> Reply to this email directly or view it on GitHub.

Commit:
coordinate descent

Commit:
reordered

zhengqu commented:
Dear Dr. Hennig,

I added a series of recent work regarding the randomized coordinate descent methods for solving large-scale convex composite minimization problems, including the serial sampling (ICDM), parallel uniform sampling (PCDM), the distributed sampling (Hydra), the accelerated parallel proximal extension (APPROX), the distributed and accelerated variant (Hydra2) and finally a unified algorithm (ALPHA) which includes all the previous algorithms as special cases. I also added a recent work on stochastic gradient method (S2GD) and a randomized dual ascent method with arbitrary sampling (Quartz).

I think that these papers are relevant to the Probabilistic Numerics community, for that they are all stochastic algorithms in nature and they are developed for tackling numerical problems in large scale setting.

Best regards

Zheng Qu

philipphennig commented:
Dear Zheng Qu,

thanks for your reply. I think this is a misunderstanding. What we mean by probabilistic numerical algorithms differs from stochastic algorithms (which, as you point out with your submission, have been studied widely). Quite in the contrary, a probabilistic numerical method typically removes random numbers from the computation as far as possible. Probabilistic numerics is specifically also (though not exclusively) interested in the use of uncertainty in wholly deterministic computations. 

One basic example that perhaps helps clarify: Consider the problem of minimizing a quadratic function f(x) = x' \* A \* x - x' \* b (aka. solving the linear problem Ax=b). Let's say A is a large symmetric positive definite matrix and b is a correspondingly large vector. There is no need for stochasticity here, it's an entirely deterministic, convex computation, with a unique solution. But if the dimensionality is very large, you still can't compute the solution in closed form, so one can ask wether it is possible to be quantitatively uncertain about it. Let's say you're also interested in the inverse of A. If you make a few conjugate gradient steps, how much do they tell you about the inverse of A? These are the kind of questions we are trying to address with a probabilistic formulation of numerical methods. This doesn't mean that stochasticity never plays a role, but to my knowledge, stochastic optimization does not answer them (though it may well contribute interesting insights). Stochastic optimization is at best tangential to probabilistic numerics.

Hence, I hope you will understand that we are reluctant to include general stochastic optimization methods in our literature list. There are so many of these algorithms that they would drown out what we actually want to discuss. A place that might be better suited for your work is the Stochastic Programming Bibliography, at http://www.eco.rug.nl/mally/spbib.html, or you could submit your work to optimization online, at http://www.optimization-online.org/cgi-bin/create.cgi (they have an eprint area for stochastic programming). 

Best wishes, and thanks again for your interest in our community. 

Philipp

> On 26 Feb 2015, at 19:46, zhengqu notifications@github.com wrote:
> 
> Dear Dr. Hennig,
> 
> I added a series of recent work regarding the randomized coordinate descent methods for solving large-scale convex composite minimization problems, including the serial sampling (ICDM), parallel uniform sampling (PCDM), the distributed sampling (Hydra), the accelerated parallel proximal extension (APPROX), the distributed and accelerated variant (Hydra2) and finally a unified algorithm (ALPHA) which includes all the previous algorithms as special cases. I also added a recent work on stochastic gradient method (S2GD) and a randomized dual ascent method with arbitrary sampling (Quartz).
> 
> I think that these papers are relevant to the Probabilistic Numerics community, for that they are all stochastic algorithms in nature and they are developed for tackling numerical problems in large scale setting.
> 
> Best regards
> 
> Zheng Qu
> 
> —
> Reply to this email directly or view it on GitHub.
